--- ntfsprogs-1.8.2/include/ntfs/endians.h.orig	2004-01-07 10:43:11.000000000 +0000
+++ ntfsprogs-1.8.2/include/ntfs/endians.h	2004-01-11 21:47:16.000000000 +0000
@@ -87,6 +87,27 @@
 #define __cpu_to_le32(x) bswap_32(x)
 #define __cpu_to_le64(x) bswap_64(x)
 
+#ifndef __bswap_constant_16
+#define __bswap_constant_16(x) \
+	((((x) >> 8) & 0xff) | (((x) & 0xff) << 8))
+#endif
+#ifndef __bswap_constant_32
+#define __bswap_constant_32(x) \
+	((((x) & 0xff000000) >> 24) | (((x) & 0x00ff0000) >>  8) | \
+	 (((x) & 0x0000ff00) <<  8) | (((x) & 0x000000ff) << 24))
+#endif
+#ifndef __bswap_constant_64
+# define __bswap_constant_64(x) \
+     ((((x) & 0xff00000000000000ull) >> 56)				      \
+      | (((x) & 0x00ff000000000000ull) >> 40)				      \
+      | (((x) & 0x0000ff0000000000ull) >> 24)				      \
+      | (((x) & 0x000000ff00000000ull) >> 8)				      \
+      | (((x) & 0x00000000ff000000ull) << 8)				      \
+      | (((x) & 0x0000000000ff0000ull) << 24)				      \
+      | (((x) & 0x000000000000ff00ull) << 40)				      \
+      | (((x) & 0x00000000000000ffull) << 56))
+#endif
+
 #define __constant_le16_to_cpu(x) __bswap_constant_16(x)
 #define __constant_le32_to_cpu(x) __bswap_constant_32(x)
 #define __constant_le64_to_cpu(x) __bswap_constant_64(x)
