--- ntfsprogs-1.7.1/libntfs/lcnalloc.c.orig	Thu Feb  6 16:24:56 2003
+++ ntfsprogs-1.7.1/libntfs/lcnalloc.c	Fri Jun 27 11:28:27 2003
@@ -101,8 +101,8 @@
 	int err = 0, rlpos, rlsize, buf_size;
 	u8 pass, done_zones, search_zone, need_writeback, bit;
 
-	Dprintf("%s(): Entering with count = 0x%Lx, start_lcn = 0x%Lx,
-			zone = %s_ZONE.\n", (long long)count,
+	Dprintf("%s(): Entering with count = 0x%Lx, start_lcn = 0x%Lx, "
+			"zone = %s_ZONE.\n", (long long)count,
 			(long long)start_lcn,
 			zone == MFT_ZONE ? "MFT" : "DATA");
 	if (!vol || count < 0 || start_lcn < -1 || !vol->lcnbmp_na ||
